### PR TITLE
import addresses when importing token balances

### LINKED ISSUE #1007: (foreign_key_violation) - Indexer.TokenBalance.Fetcher
```
14:14:35.024 [error] Task #PID<0.17008.92> started from Indexer.TokenBalance.Fetcher terminating
** (Postgrex.Error) ERROR 23503 (foreign_key_violation): insert or update on table "address_token_balances" violates foreign key constraint "address_token_balances_address_hash_fkey"

    table: address_token_balances
    constraint: address_token_balances_address_hash_fkey

Key (address_hash)=(\x01a292d4cf8d9329f7a63b5b030467a0e347eb0c) is not present in table "addresses".
    (explorer) lib/explorer/repo.ex:58: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
    (elixir) lib/enum.ex:1925: Enum."-reduce/3-lists^foldl/2-0-"/3
    (explorer) lib/explorer/chain/import.ex:303: Explorer.Chain.Import.insert_changes_list/2
    (explorer) lib/explorer/chain/import/address/token_balances.ex:66: Explorer.Chain.Import.Address.TokenBalances.insert/2
    (ecto) lib/ecto/multi.ex:421: Ecto.Multi.apply_operation/5
    (elixir) lib/enum.ex:1925: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ecto) lib/ecto/multi.ex:411: anonymous fn/5 in Ecto.Multi.apply_operations/5
    (ecto) lib/ecto/adapters/sql.ex:576: anonymous fn/3 in Ecto.Adapters.SQL.do_transaction/3
Function: &Indexer.TokenBalance.Fetcher.run/2
    Args: [[{<<164, 191, 92, 204, 227, 192, 127, 122, 83, 157, 117, 101, 153, 233, 224, 21, 212, 75, 254, 167>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<172, 66, 220, 43, 20, 120, 14, 102, 61, 142, 50, 15, 216, 142, 178, 186, 27, 6, 150, 24>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<175, 200, 76, 182, 228, 3, 138, 137, 175, 19, 201, 177, 217, 238, 44, 228, 22, 134, 91, 131>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<177, 117, 208, 239, 91, 74, 189, 63, 126, 37, 233, 175, 86, 194, 192, 24, 176, 107, 204, 202>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<181, 50, 35, 68, 6, 162, 251, 70, 183, 116, 100, 182, 252, 18, 85, 24, 214, 185, 48, 198>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<198, 205, 134, 121, 223, 192, 141, 180, 72, 96, 186, 119, 58, 123, 38, 15, 68, 231, 56, 84>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<224, 35, 249, 130, 201, 124, 231, 154, 127, 188, 40, 138, 44, 4, 28, 173, 247, 30, 9, 153>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<227, 100, 229, 193, 236, 150, 55, 3, 160, 210, 91, 151, 132, 44, 60, 251, 44, 191, 91, 129>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<230, 124, 234, 166, 110, 91, 204, 101, 187, 29, 12, 97, 122, 183, 166, 236, 12, 92, 132, 75>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<232, 135, 161, 101, 104, 66, 130, 112, 81, 210, 74, 142, 232, 192, 221, 107, 89, 203, 64, 18>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304874}, {<<12, 36, 79, 110, 36, 190, 146, 3, 215, 197, 47, 155, 174, 106, 52, 58, 53, 244, 141, 192>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<78, 139, 174, 231, 239, 127, 98, 40, 194, 247, 40, 211, 202, 70, 216, 6, 247, 180, 17, 190>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304856}, {<<125, 25, 82, 166, 238, 202, 221, 160, 213, 43, 139, 72, 70, 229, 203, 4, 102, 206, 183, 180>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304851}, {<<131, 99, 252, 220, 151, 159, 117, 111, 166, 63, 0, 250, 75, 173, 73, 100, 111, 233, 39, 52>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304851}, {<<80, 172, 180, 236, 164, 78, 241, 63, 179, 107, 212, 49, 61, 162, 170, 218, 187, 3, 200, 75>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304856}, {<<146, 211, 4, 42, 120, 21, 78, 5, 52, 70, 171, 223, 54, 136, 49, 149, 34, 117, 162, 59>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<158, 183, 58, 129, 110, 192, 5, 187, 117, 29, 23, 251, 156, 215, 130, 169, 200, 21, 232, 163>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<170, 210, 18, 197, 150, 137, 245, 44, 115, 201, 213, 182, 28, 181, 140, 206, 214, 58, 77, 138>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304856}, {<<97, 143, 173, 170, 222, 139, 211, 234, 49, 45, 201, 94, 172, 64, 123, 212, 60, 212, 251, 93>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<209, 234, 107, 129, 178, 96, 252, 194, 102, 228, 38, 88, 215, 177, 102, 252, 144, 230, 229, 33>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304851}, {<<98, 77, 241, 81, 49, 180, 63, 66, 213, 166, 110, 250, 32, 76, 29, 188, 120, 38, 155, 184>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<209, 199, 245, 76, 123, 0, 16, 176, 16, 81, 93, 220, 113, 201, 6, 28, 209, 218, 89, 137>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<3, 129, 45, 162, 213, 36, 234, 205, 230, 88, 130, 233, 15, 156, 68, 194, 10, 187, 197, 127>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304851}, {<<175, 196, 92, 5, 128, 22, 131, 113, 228, 98, 72, 192, 193, 118, 137, 193, 26, 125, 41, 241>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304851}, {<<31, 240, 91, 4, 6, 96, 66, 206, 114, 214, 85, 0, 167, 7, 78, 137, 231, 54, 110, 153>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304856}, {<<142, 230, 202, 134, 163, 117, 73, 74, 3, 86, 68, 119, 64, 150, 194, 218, 88, 124, 142, 246>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304853}, {<<75, 134, 137, 183, 91, 128, 199, 250, 100, 241, 35, 14, 30, 225, 160, 84, 26, 84, 215, 23>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, 235>>, 4304851}, {<<201, 3, 213, 50, 120, 243, 154, 157, 192, 53, 11, 244, 32, 244, 48, 250, 193, 17, 104, 167>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, 61, ...>>, 4304856}, {<<124, 106, 21, 148, 157, 129, 152, 42, 190, 208, 43, 171, 190, 42, 247, 235, 200, 178, 149, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, 9, ...>>, 4304856}, {<<160, 114, 49, 68, 189, 109, 143, 100, 102, 168, 192, 103, 114, 196, 113, 237, 151, 118, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, 202, ...>>, 4304856}, {<<205, 176, 122, 233, 172, 7, 254, 19, 41, 240, 126, 205, 125, 211, 131, 109, 4, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, 71, ...>>, 4304856}, {<<85, 167, 21, 11, 167, 77, 254, 42, 178, 9, 7, 94, 222, 228, 149, 138, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, 159, ...>>, 4304856}, {<<194, 24, 148, 125, 238, 147, 35, 59, 164, 177, 124, 39, 23, 172, 247, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, 98, ...>>, 4304856}, {<<140, 98, 121, 254, 97, 254, 169, 78, 89, 246, 58, 21, 106, 243, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, 177, ...>>, 4304856}, {<<61, 36, 156, 238, 133, 91, 139, 9, 98, 52, 133, 25, 53, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, 162, ...>>, 4304856}, {<<77, 220, 39, 75, 66, 31, 107, 221, 123, 161, 159, 19, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, 172, ...>>, 4304856}, {<<118, 61, 159, 249, 124, 135, 198, 50, 185, 85, 162, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, 174, ...>>, 4304856}, {<<221, 18, 78, 110, 104, 20, 145, 24, 26, 241, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, 49, ...>>, 4304856}, {<<142, 90, 122, 81, 201, 94, 70, 223, 68, ...>>, <<222, 135, 62, 7, 28, 144, 70, 195, ...>>, 4304856}, {<<74, 90, 127, 46, 190, 184, 205, 18, ...>>, <<222, 135, 62, 7, 28, 144, 70, ...>>, 4304856}, {<<68, 7, 120, 85, 81, 81, 35, ...>>, <<222, 135, 62, 7, 28, 144, ...>>, 4304856}, {<<225, 221, 60, 49, 105, 13, ...>>, <<222, 135, 62, 7, 28, ...>>, 4304856}, {<<12, 74, 158, 85, 108, ...>>, <<222, 135, 62, 7, ...>>, 4304856}, {<<145, 115, 197, 25, ...>>, <<222, 135, 62, ...>>, 4304856}, {<<194, 237, 79, ...>>, <<222, 135, ...>>, 4304856}, {<<198, 228, ...>>, <<222, ...>>, 4304856}, {<<163, ...>>, <<...>>, ...}, {<<...>>, ...}, {...}, ...], [transport: EthereumJSONRPC.HTTP, transport_options: [http: EthereumJSONRPC.HTTP.HTTPoison, url: "**redacted**", method_to_url: [eth_getBalance: "**redacted**", trace_block: "**redacted**", trace_replayTransaction: "**redacted**"], http_options: [recv_timeout: 60000, timeout: 60000, hackney: [pool: :ethereum_jsonrpc]]], variant: EthereumJSONRPC.Parity]]
```

### Environment

* Operating System: Linux
* Client: Parity
* Chain: Ropsten

### Steps to reproduce
1. I believe this address has a `0` balances but received a token transfer therefore there is no mention of this address in the address table.

### PR DESCRIPTION
**Pull request by GuSilverFlame**
closes #1007 
### Bug Fixes
  
- import addresses along with token balances in case an address wasn't indexed yet for some reason

### CODE REVIEW

File: apps/indexer/test/indexer/token_balance/fetcher_test.exs [hunk 1]
```diff
@@ -87,5 +87,22 @@ defmodule Indexer.TokenBalance.FetcherTest do
 
       assert TokenBalance.Fetcher.import_token_balances(token_balances_params) == :error
     end
+
+    test "insert ok when the address does not exist yet" do
+      contract = insert(:token)
+      insert(:block, number: 19999)
+
+      Explorer.Chain
```
Comment (amandasposito) [inline apps/indexer/test/indexer/token_balance/fetcher_test.exs]: Why do we need this for?

File: apps/indexer/test/indexer/token_balance/fetcher_test.exs [hunk 2]
```diff
@@ -87,5 +87,22 @@ defmodule Indexer.TokenBalance.FetcherTest do
 
       assert TokenBalance.Fetcher.import_token_balances(token_balances_params) == :error
     end
+
+    test "insert ok when the address does not exist yet" do
```
Comment (amandasposito) [inline apps/indexer/test/indexer/token_balance/fetcher_test.exs]: WDYT of a better test description? What happens here to return `ok`?

File: apps/indexer/test/indexer/token_balance/fetcher_test.exs (hunk 1 above)
Comment (GuSilverFlame) [reply apps/indexer/test/indexer/token_balance/fetcher_test.exs]: we don't, it's a leftover from something I was experimenting with...
I'll remove it now

File: apps/indexer/test/indexer/token_balance/fetcher_test.exs [hunk 3]
```diff
@@ -87,5 +87,20 @@ defmodule Indexer.TokenBalance.FetcherTest do
...
+
+      token_balances_params = [
+        %{
+          address_hash: "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+          block_number: 19999,
+          token_contract_address_hash: to_string(contract.contract_address_hash)
+        }
+      ]
+
+      assert TokenBalance.Fetcher.import_token_balances(token_balances_params) == :ok
```
Comment (amandasposito) [inline apps/indexer/test/indexer/token_balance/fetcher_test.exs]: WDYT of an `assert` to verify if the `address` was inserted?

Comment (GuSilverFlame) [reply apps/indexer/test/indexer/token_balance/fetcher_test.exs]: done

File: apps/indexer/lib/indexer/token_balance/fetcher.ex [hunk 4]
```diff
@@ -80,7 +80,12 @@ defmodule Indexer.TokenBalance.Fetcher do
   end
 
   def import_token_balances(token_balances_params) do
+    addresses_params =
+      Enum.map(token_balances_params, fn %{address_hash: hash} -> %{hash: hash} end)
```
Comment (feliperenan) [inline apps/indexer/lib/indexer/token_balance/fetcher.ex]: If we apply this `uniq` we can change it to:

```Elixir
token_balances_params
|> Enum.map(& &1.address_hash)
|> Enum.uniq()
```

Maybe move it to a private function 🤔.

**Comment by amandasposito** [comment]
I've been indexing the Sokol chain for the last couple of hours and no errors related to the foreign key appeared. 👍

**Pull request merged**